src/generic/MPoly.jl

```diff
@@ -2409,11 +2427,14 @@ function ^(a::MPoly{T}, b::Int) where {T <: RingElement}
       end
       return parent(a)([coeff(a, 1)^b], exps)
    elseif b == 0
-      return parent(a)(1)
+      return one(a)
    elseif b == 1
-      return a
+      return deepcopy(a)
    elseif b == 2
       return a*a
+   elseif characteristic(base_ring(a)) != 0
```

Comment:
This won't work if the characteristic is `missing`, as `missing != 0` is equal to `missing`, which is not allowed in a condition. You must write something like `if coalesce(characteristic(base_ring(a)), -1) != 0`.

Reply:
I will look into adding a test for that, with a ring which uses the `missing` fallback.

Reply:
```suggestion
   elseif coalesce(characteristic(base_ring(a)), -1) != 0
```
Made a "code suggestion" to make it easier to include.

Reply:
This seems to be equivalent to just returning -1 instead of missing, which would be a lot simpler. I think I will do that, as this seems to be a lot messier in comparison. Of course, -1 cannot be a valid characteristic and it can just stand for "don't know".

Reply:
Yes I agree, `missing` is a more explicit sentinel, hence my earlier suggestion, but it's simple `-1`. The drawback is that the code might silently fail if we forget to check against `-1`, but tests should help. Plus, it's internal only (for the moment) so we don't need to think too much about the long term consequences of this choice.